symengine/lib/symengine_wrapper.pyx

```diff
@@ -1161,6 +1178,46 @@ cdef class Complex(Number):
         import sage.all as sage
         return self.real_part()._sage_() + sage.I * self.imaginary_part()._sage_()
 
+cdef class Infinity(Number):
+
+    def _sympy_(self):
+        import sympy
+        return sympy.oo
+
+    def _sage_(self):
+        import sage.all as sage
+        return sage.oo
+
+cdef class NegativeInfinity(Number):
+
+    def _sympy_(self):
+        import sympy
+        return -sympy.oo
+
+    def _sage_(self):
+        import sage.all as sage
+        return -sage.oo
+
+cdef class ComplexInfinity(Number):
+
+    def _sympy_(self):
+        import sympy
+        return sympy.zoo
+
+    def _sage_(self):
+        import sage.all as sage
+        return sage.unsigned_infinity
+
+cdef class NaN(Number):
```

Comment:
See how I've handled `Add`. You just need to change c2py `NaN.__new__(NaN)` to `Basic.__new__(NaN)`